### ISSUE TITLE
Windows "previous versions" now opens pwsh.exe not explorer

### ISSUE THREAD

**Issue by Jackbennett** [PowerShell/PowerShell#6799]
Steps to reproduce
------------------
1. Install the preview MSI
2. Click Open at a shadow copy prior date;

![image](https://user-images.githubusercontent.com/791134/39518106-c80ce8c0-4df9-11e8-9cbd-8b7ef94c674f.png)

Expected behavior
-----------------
Windows explorer at this file path.

Actual behavior
---------------
pwsh.exe at path `PS V:\@GMT-2018.04.23-10.11.03\` (of the time opened)

Work Around
--------------
I can just type `start .` to get explorer

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.2
PSEdition                      Core
GitCommitId                    v6.1.0-preview.2
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by GeeLaw** [comment]
Did you mean installing the preview PowerShell somehow makes File Explorer think `pwsh.exe` is the shell program (which is usually `explorer.exe`)?

**Comment by Jackbennett** [comment]
Yeh but it's not like win-e does anything different. I'm not sure what
handler would have changed.

**Comment by bergmeister** [comment]
@Jackbennett Does this still happen with `6.1.0` in your environment?

**Comment by Jackbennett** [comment]
I'm afraid to report the same issue

![image](https://user-images.githubusercontent.com/791134/46342429-9a0ce180-c632-11e8-8b06-58a7a9e6f2c9.png)


Sorry it's been a while but on the up side this is completely re-imaged with MDT 1709 windows 10 x64 which was a fresh build from a microsoft ISO install from the initial report.

# Steps to reproduce
1. Find a folder with previous versions
1. Select "open" on an older folder version
1. You will see explorer open at that point in time.
1. Install `PowerShell-6.1.0-win-x64.msi`
1. Click open again, You now get pwsh.exe at the previous version file path

```
> $PsversionTable
Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

Winver: 
![image](https://user-images.githubusercontent.com/791134/46342626-2b7c5380-c633-11e8-8df7-5ac949a3bd65.png)

**Comment by bergmeister** [comment]
@Jackbennett Thanks for the detailed report. I suspect this might be related to #7815 that I started to investigate yesterday. I hope to have an answer in the next days. For the moment the recommendation is to re-install PowerShell without the context menu option if this is a blocking problem.

**Comment by Jackbennett** [comment]
For completeness removed/installed without the context option ticked does indeed fix the issue. Interesting  coincidence.

**Comment by bergmeister** [comment]
Thanks for confirming this (as it proves the problem is caused by this feature). I will investigate but there is a chance that this might even be a bug in Windows

**Comment by GeeLaw** [comment]
Perhaps it is because PowerShell context menu assumed the default verb for folders?

**Comment by bergmeister** [comment]
@GeeLaw Where did you see that we assume the default verb for folders?
I checked all the registry keys and could boil it down to the `HKCR\Directory\ContextMenus\PowerShell6x64\shell\open` [here](https://github.com/PowerShell/PowerShell/blob/master/assets/Product.wxs#L209), which is necessary for non-admin sub menu context menu.

**Comment by GeeLaw** [comment]
@bergmeister Sorry, it isn't because PS6 assumes the **default** verb, it is because it takes the `open` verb -- the cascaded menu verb is named as `open`. If `SEE_MASK_INVOKEIDLIST` is used, this verb will be found by `ShellExecuteEx`, which is the case for "Open" in "Previous Versions".

If you rename `HKCR\Directory\ContextMenus\PowerShell6x64\shell\open` to `openpwsh`, the problem will go away.

Ordinarily, the `open` verb is found in `HKLM\SOFTWARE\Classes\Folder\shell\open`. PowerShell's context menu overrides this because it is found **later** than the should-be `open`.

I'm writing a blog entry for a detailed analysis on this issue. Will be modifying this comment to reflect the entry later.

It is also worth mentioning that currently the installer does **not** install on a per-user basis. The cascaded context menu will always be created for all users. As a piece of advice, the installer should be more specific on whether to install the keys to `HKCU\Software\Classes` or `HKLM\SOFTWARE\Classes`. Even if the installer is running as administrator, the value still might be written to `HKCU\Software\Classes`, if the key already exists in `HKCU`.

**Comment by bergmeister** [comment]
@GeeLaw Thanks for the detailed answer, I opened a pull request with a fix for the `open` verb to use `openpwsh` but I don't think we should change `runas` to `runaspwsh` due to the required elevation (and having a flickering launcher instead is not preferred instead, I proposed this already when I originally did the feature) because as far as I can see, continuing to use the default verb for `runas` should not cause a problem or is probably an edge case to be disregarded.

**Comment by JHBoricua** [comment]
Has there been a regression of this? I just installed the latest Powershell Core MSI (PowerShell-6.1.3-win-x64) and it's opening the previous version of a folder in a powershell window. Uninstalling PowerShell core reverts the behavior back to normal.

Or has something changed and this is the intended behavior now?

**Comment by bergmeister** [comment]
@JHBoricua The fix was in https://github.com/PowerShell/PowerShell/pull/7932/files You can just check your registry entries. The fix was made after 6.1 was released, therefore it is very unlikely, the commit was included in one of the patches (which are  mostly just for security critical patching of the runtime, etc.). Please try 6.2-rc

**Comment by kryptowastaken** [comment]
This problem is back in PowerShell 7 v7.3.7, the registry is correctly what it is supoosed to be but sill previous versions are opening in pwsh.exe when clicking the open button, however when double-clicking the folder itself, the previous version opens in File Explorer
![image](https://github.com/PowerShell/PowerShell/assets/141824011/c7bb17c4-8ba1-4c32-99cd-341fef03ce86)
